src/db/users.rs

```diff
@@ -0,0 +1,60 @@
+use crate::github::User;
+use anyhow::Context;
+use tokio_postgres::Client as DbClient;
+
+/// Add a new user.
+/// If an user already exists, updates their username.
+pub async fn record_username(db: &DbClient, user_id: u64, username: &str) -> anyhow::Result<()> {
+    db.execute(
+        r"
+INSERT INTO users (user_id, username) VALUES ($1, $2)
+ON CONFLICT (user_id)
+DO UPDATE SET username = $2",
+        &[&(user_id as i64), &username],
+    )
+    .await
+    .context("inserting user id / username")?;
+    Ok(())
+}
+
+/// Return a user from the DB.
+pub async fn get_user(db: &DbClient, user_id: u64) -> anyhow::Result<Option<User>> {
+    let row = db
+        .query_opt(
+            r"
+SELECT username
+FROM users
+WHERE user_id = $1;",
+            &[&(user_id as i64)],
+        )
+        .await
+        .context("cannot load user from DB")?;
+    Ok(row.map(|row| {
+        let username: &str = row.get(0);
+        User {
+            id: user_id,
+            login: username.to_string(),
+        }
+    }))
```

Comment:
I think if you `SELECT *` then you can just `Ok(row.into())` but it's a matter of preference ...

Reply:
We don't currently implement `From<tokio_postgres::Row> for User` anywhere :( Which goes back to my comment that we kind of misuse the GitHub user to also mean the database user.

Introducing a separate type would mean doing some conversions, but I guess that if we want to keep the `users` table, it's the Right Thing™ to do, in the long term. What do you think? :)